src/cli/templates/decorators/with-route-segment-config.ts

```diff
@@ -1,82 +1,54 @@
-import {
-  type CompileFn,
-  type DecoratorParams,
-  getPattern,
-  getPatternsFromNames,
-} from '../tpl-utils'
+import { type CompileFn, type DecoratorParams, getPattern } from '../tpl-utils'
 
-const PATTERNS = getPatternsFromNames(
+const CONFIG_VARIABLES = [
   'dynamic',
   'dynamicParams',
   'revalidate',
   'fetchCache',
   'runtime',
-  'preferredRegion'
-)
+  'preferredRegion',
+] as const
 
-type ConfigVariable = keyof typeof PATTERNS
-
-const TEMPLATES: Record<ConfigVariable, string> = {
-  dynamic: `export const dynamic = ${PATTERNS.dynamic}
-`,
-  dynamicParams: `export const dynamicParams = ${PATTERNS.dynamicParams}
-`,
-  revalidate: `export const revalidate = ${PATTERNS.revalidate}
-`,
-  fetchCache: `export const fetchCache = ${PATTERNS.fetchCache}
-`,
-  runtime: `export const runtime = ${PATTERNS.runtime}
-`,
-  preferredRegion: `export const preferredRegion = ${PATTERNS.preferredRegion}
-`,
-}
+type ConfigVariable = (typeof CONFIG_VARIABLES)[number]
 
 const REG_EXPS: Record<ConfigVariable, RegExp> = {
-  dynamic: /export const dynamic = (.*)/,
-  dynamicParams: /export const dynamicParams = (.*)/,
-  revalidate: /export const revalidate = (.*)/,
-  fetchCache: /export const fetchCache = (.*)/,
-  runtime: /export const runtime = (.*)/,
-  preferredRegion: /export const preferredRegion = (.*)/,
-}
-
-function getCompileParams(input: string) {
-  return Object.entries(REG_EXPS).reduce(
-    (params, [variable, regExp]) => {
-      params[variable as ConfigVariable] = input.match(regExp)?.[1]
-      return params
-    },
-    {} as Record<ConfigVariable, string | undefined>
-  )
+  dynamic: /export .+ dynamic/,
+  dynamicParams: /export .+ dynamicParams/,
+  revalidate: /export .+ revalidate/,
+  fetchCache: /export .+ fetchCache/,
+  runtime: /export .+ runtime/,
+  preferredRegion: /export .+ preferredRegion/,
 }
 
-function compileConfigVariableTemplate(
-  variable: ConfigVariable,
-  value: string
-) {
-  return TEMPLATES[variable].replace(getPattern(variable), value)
+function getExistingConfigVariables(input: string): ConfigVariable[] {
+  return CONFIG_VARIABLES.filter((variable) => REG_EXPS[variable].test(input))
 }
 
-function compileConfigTemplateFactory(
-  params: Record<ConfigVariable, string | undefined>
+function createReExportTemplate(
+  configVariables: ConfigVariable[],
+  originPath: string
 ) {
-  const config = Object.entries(params).reduce((output, [variable, value]) => {
-    if (!value) return output
-    return `${output}${compileConfigVariableTemplate(
-      variable as ConfigVariable,
-      value
-    )}`
-  }, '')
-  return (input: string) =>
-    config === ''
-      ? input
-      : `${input}
-${config}`
+  if (configVariables.length === 0) {
+    return ''
+  }
+
+  const exportList = configVariables.join(', ')
+  return `
+export { ${exportList} } from '${originPath}'
```

Comment:
Nextjs docs says:

"The values of the config options currently need be statically analyzable. For example revalidate = 600 is valid, but revalidate = 60 * 10 is not."

Are we ok here?